RenderAreaWidget.py

```diff
@@ -60,16 +63,57 @@ def mouseMoveEvent(self, event):
     def mouseReleaseEvent(self, event):
         self.__selected = None
 
+    #def filePresent(self, fileName):
+    #    try:
+    #        if os.path.exist(fileName):
+    #            print ("print file present - can be used")
+    #            return True
+    #    except:
+    #        print ("json file does not exist, creation to do")
+    #        return False
+
+    def finalDataShape(self, predatashape, tojson, currentIndex):
+        """goal: finalize the shape data before to save it in the json file """
+        temp = {}
+        posElement = 'element' + str(currentIndex)
+        # copy of dictionnary
+        temp[posElement] = predatashape.copy()
+        locationValue = predatashape['location']
+        if locationValue not in tojson.keys():
+            tojson[locationValue]= []
+            tojson[locationValue].append(temp)
+        else:
+            tojson[locationValue].append(temp)
+        return tojson
+
+    def toJson(self):
+        """will save the different elements present in __shape to a json files"""
+        fileName= "first_save.json"
+        tojson = {}
+        # add a test if the file exist or not
+        # if exist add a different opening file to append more data
+        with open(fileName, 'w') as f:
+            currentIndex = 0
+            for shape in self.__shapes:
+                predatashape = shape.predataShape()
+                tojson = self.finalDataShape(predatashape, tojson, currentIndex)
```

Comment:
remove egality, tojson is mutable 

a = 1
b = a
b +=  1
=> b = 2 et a = 1

l = [1, 2]
k = l
k[1] = 5
=> l = [5, 2], k = [5, 2]

par argument d'une function, immutable est utilisatble sans changer la valeur original
def fun(a):
  a = 2

a = 1
fun(a)
=> a = 1

def fun(l):
  l.append("toto")

l = [1, 2]
fun(l)
==> l = [1, 2, "toto"]

self.finalDataShape(tosjon)